apix-impl/src/integration-test/java/eu/xenit/apix/tests/search/SearchServiceTest.java

```diff
@@ -438,4 +438,30 @@ public void invalidLong_onlyNode_doesNotThrowException() {
         searchService.query(searchQuery);
     }
 
+    @Test
+    public void testSearchForQuotationMarkProperty() {
```

Comment:
What is being tested here? Syntax is already tested in unit test above. Integration with Solr is tested in other integration tests. So we are actually testing Solr itself here, which I think is not needed.

Reply:
The test confirms that you can actually find a document with a quotation mark in it. The previous test suite unit-tested the syntax, but due to the lack of an integration test didn't notice that the test was incorrect/insufficient.The unit test doesn't entirely prevent regressions, since an abstraction layer can easily be added that interprets backslashes an additional time before the query is actually executed. The integration test would catch that.

The query is transactional so it doesn't need to wait for Solr, meaning that unlike some other search tests, it doesn't increase the build time by several seconds.

Reply:
You have a good argument (although it can be applied to (too) many tests, and at a certain point you have to start pruning) but it's ok for me to keep it in.